src/ResourceManager/PowerBI/Commands.PowerBI/Commands/UpdateAzureRmPowerBIEmbeddedCapacity.cs

```diff
@@ -0,0 +1,144 @@
+using Microsoft.Azure.Commands.PowerBI.Properties;
+using Microsoft.Azure.Commands.PowerBI.Utilities;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.PowerBIDedicated.Models;
+
+namespace Microsoft.Azure.Commands.PowerBI
+{
+    [Cmdlet(VerbsData.Update, "AzureRmPowerBIEmbeddedCapacity", SupportsShouldProcess = true), 
+        OutputType(typeof(PSDedicatedCapacity))]
+    public class UpdateAzurePowerBIEmbeddedCapacity : PowerBICmdletBase
+    {
+        protected const string CmdletParametersSet = "ByNameAndResourceGroup";
+        protected const string ObjectParameterSet = "ByInputObject";
+        protected const string ResourceIdParameterSet = "ByResourceId";
+
+        private const string ParamSetDefault = "Default";
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Position = 0, 
+            Mandatory = true,
+            HelpMessage = "Name of the capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = false,
+            HelpMessage = "Name of resource group under which you want to update the capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Name of the Sku used to create the capacity")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("A1", "A2", "A3", "A4", "A5", "A6", IgnoreCase = true)]
+        public string Sku { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A string,string dictionary of tags associated with this capacity")]
+        [ValidateNotNull]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A comma separated capacity names to set as administrators on the capacity")]
+        [ValidateNotNull]
+        public string[] Administrator { get; set; }
+
+        [Parameter(
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "PowerBI Embedded Capacity ResourceID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ObjectParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "PowerBI Embedded Capacity object.")]
+        [ValidateNotNullOrEmpty]
+        public PSDedicatedCapacity InputObject { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string capacityName = string.Empty;
```

Comment:
@selasagi same comment about setting this to `Name` and removing the below if-statement

Reply:
fixed